app/models/replication_transfer.rb

```diff
@@ -21,10 +21,10 @@ def self.find_fields
 
 
 
-  def cancel!(reason)
+  def cancel!(reason, detail)
     unless cancelled
       transaction do
-        update!(cancelled: true, cancel_reason: reason)
+        update!(cancelled: true, cancel_reason: reason, cancel_reason_detail: detail)
         bag_man_request&.cancel!(reason)
       end
     end
```

Comment:
I agree there must be a better strategy.  The root of the complexity seems to be the coupling between the concerns, and the fact that the concern addressed by BagManRequest are addressed via external tools at some nodes.  

Moving this into the controller level offers a fair amount of room to maneuver, but would require rewriting the Repl jobs to use the new route.  I think we'll want to look at this for a minor release soon.